lib/ASTGen/Sources/ASTGen/Attrs.swift

```diff
@@ -0,0 +1,1132 @@
+      arguments.count >= 1
+    else {
+      // TODO: Diagnose?
+      return nil
+    }
+    let effectKind: BridgedEffectsKind
+    switch arguments.first!.rawText {
+    case "readonly":
+      effectKind = .readOnly
+    case "readnone":
+      effectKind = .readNone
+    case "releasenone":
+      effectKind = .releaseNone
+    case "readwrite":
+      effectKind = .readWrite
+    default:
+      effectKind = .custom
+    }
+
+    if effectKind != .custom {
+      return .createParsed(
+        self.ctx,
+        atLoc: self.generateSourceLoc(node.atSign),
+        range: self.generateSourceRange(node),
+        effectKind: effectKind
+      )
+    } else {
+      return .createParsed(
+        self.ctx,
+        atLoc: self.generateSourceLoc(node.atSign),
+        range: self.generateSourceRange(node),
+        customString: self.extractRawText(arguments).bridged,
+        customStringLoc: self.generateSourceLoc(arguments)
+      )
+    }
+  }
+
+  func generateExclusivityAttr(attribute node: AttributeSyntax) -> BridgedExclusivityAttr? {
+    let mode: BridgedExclusivityAttrMode? = self.generateSingleAttrOption(
+      attribute: node,
+      {
+        switch $0.rawText {
+        case "checked": return .checked
+        case "unchecked": return .unchecked
+        default: return nil
+        }
+      }
+    )
+    guard let mode else {
+      return nil
+    }
+    return .createParsed(
+      self.ctx,
+      atLoc: self.generateSourceLoc(node.atSign),
+      range: self.generateSourceRange(node),
+      mode: mode
+    )
+  }
+
+  func generateExposeAttr(attribute node: AttributeSyntax) -> BridgedExposeAttr? {
+    guard
+      var args = node.arguments?.as(LabeledExprListSyntax.self)?[...]
+    else {
+      // TODO: Diagnose.
+      return nil
+    }
+
+    // Exposure kind.
+    let kind: BridgedExposureKind? = self.generateConsumingPlainIdentifierAttrOption(args: &args) {
+      switch $0.rawText {
+      case "Cxx":
+        return .cxx
+      case "Wasn":
+        return .wasm
+      default:
+        return nil
+      }
+    }
+    guard let kind else {
+      return nil
+    }
+
+    // Name.
+    let name = self.generateConsumingSimpleStringLiteralAttrOption(args: &args) ?? ""
+
+    guard args.isEmpty else {
+      // TODO: Diagnose
+      return nil
+    }
+
+    return .createParsed(
+      self.ctx,
+      atLoc: self.generateSourceLoc(node.atSign),
+      range: self.generateSourceRange(node),
+      name: name,
+      kind: kind
+    )
+  }
+
+  func generateExternAttr(attribute node: AttributeSyntax) -> BridgedExternAttr? {
+    guard
+      var args = node.arguments?.as(LabeledExprListSyntax.self)?[...]
+    else {
+      // TODO: Diagnose.
+      return nil
+    }
+
+    // @_extern(wasm, module: "x", name: "y")
+    // @_extern(c[, "x"])
+    let kind: BridgedExternKind? = self.generateConsumingPlainIdentifierAttrOption(args: &args) {
+      switch $0.rawText {
+      case "c":
+        return .C
```

Comment:
Any reason why `.C` is uppercased by `.wasm` is lowercase?

Reply:
That's how C interop imports them🤷 
```
BridgedExternKindC -> .C
BridgedExternKindWasm -> .wasm
```

Reply:
OK. Whatever 🤷🏽